### PR TITLE
chore: Update ko builder to use ko v0.13.0

### PR DESCRIPTION
**Pull request by halvards**
Fixes: #7851 and #8698

The main user-facing change in ko v0.13.0 that is relevant to current Skaffold features is the change in default base image from [`gcr.io/distroless/static:nonroot`](https://github.com/GoogleContainerTools/distroless/tree/main/base) to [`cgr.dev/chainguard/static:latest`](https://github.com/chainguard-images/images/tree/main/images/static).

Some of the Skaffold ko builder examples overrides the default base image with `gcr.io/distroless/static-debian11:nonroot`. This assists users who run Skaffold in Google Cloud VPC networks with limited or no internet connectivity. These users can access Google's distroless images in [Container Registry](https://cloud.google.com/container-registry/docs/) and [Artifact Registry](https://cloud.google.com/artifact-registry/docs/) via [Private Google Access](https://cloud.google.com/vpc/docs/private-google-access).

This change also updates the ko builder maturity to Beta :-)

### CODE REVIEW

File: docs-v2/data/maturity.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
     "debug": "x",
     "area": "Build",
     "feature": "ko builder",
-    "maturity": "alpha",
+    "maturity": "beta",
```
Comment (halvards) [inline docs-v2/data/maturity.json:91]: I hope this is ok :-D. Please let me know if it's not, I'd be happy to revert it.

File: integration/examples/ko/skaffold.yaml [hunk 2]
```diff
@@ -22,8 +22,9 @@ build:
         paths:
         - "**/*.go"
         - go.*
+      fromImage: gcr.io/distroless/static-debian11:nonroot
 profiles:
 - name: gcb
   build:
     googleCloudBuild:
-      koImage: us-central1-docker.pkg.dev/k8s-skaffold/testing/skaffold:v1.37.2-lts
+      koImage: us-central1-docker.pkg.dev/k8s-skaffold/testing/skaffold:v1.37.2
```
Comment (halvards) [inline integration/examples/ko/skaffold.yaml:30]: The previous tag didn't exist in the repository.

File: integration/examples/custom/build.sh [hunk 3]
```diff
@@ -1,15 +1,14 @@
 #!/usr/bin/env bash
-set -e
-set -x
+set -Eefuo pipefail
 
 if ! [ -x "$(go env GOPATH)/bin/ko" ]; then
-    pushd $(mktemp -d)
-    curl -L https://github.com/google/ko/archive/v0.12.0.tar.gz | tar --strip-components 1 -zx
-    go build -o $(go env GOPATH)/bin/ko .
+    pushd "$(mktemp -d)"
```
Comment (halvards) [inline integration/examples/custom/build.sh:5]: Adding quotes here to fix `shellcheck` errors.

File: integration/examples/ko/cloudbuild.yaml [hunk 4]
```diff
@@ -58,7 +58,7 @@ substitutions:
   _GKE_CLUSTER_PROJECT_ID: $PROJECT_ID
   _GKE_CLUSTER_ZONE: us-central1-f
   _IMAGE_REPO: gcr.io/${PROJECT_ID}
-  _GCLOUD_IMAGE: us-central1-docker.pkg.dev/k8s-skaffold/testing/skaffold
-  _SKAFFOLD_IMAGE: us-central1-docker.pkg.dev/k8s-skaffold/testing/skaffold
+  _GCLOUD_IMAGE: gcr.io/k8s-skaffold/skaffold
```
Comment (halvards) [inline integration/examples/ko/cloudbuild.yaml:61]: The current images don't exist afaict.

File: docs-v2/data/maturity.json (hunk 1 above)
Comment (aaron-prindle) [reply docs-v2/data/maturity.json:91]: 👍

**Review by aaron-prindle — APPROVED** [review]
LGTM

**Pull request merged**